src/Vegetation/photosynthesis.jl

```diff
@@ -192,3 +221,54 @@ function compute_photosynthesis(
     Rd = dark_respiration(Vcmax25, β, f, ΔHRd, T, To, R)
     return net_photosynthesis(Ac, Aj, Rd, β)
 end
+
+function compute_Jmax25(Jmax_model::PrescribedJmax25, _...)
+    return Jmax_model.Jmax25
+end
+
+
+function compute_Jmax25(Jmax_model::OptimalityJmax25, parameters::FarquharParameters, T, medlyn_factor,
+                         APAR, c_co2, R)
+    """Solve for Jmax25 assuming Aj=Ac"""
+    (;
+        Γstar25,
+        ΔHJmax,
+        ΔHVcmax,
+        ΔHΓstar,
+        f,
+        ΔHRd,
+        To,
+        θj,
+        ϕ,
+        mechanism,
+        sc,
+        ψc,
+        oi,
+        Kc25,
+        Ko25,
+        ΔHkc,
+        ΔHko,
+        jv_ratio,
+    ) = parameters
+
+    Γstar = co2_compensation(Γstar25, ΔHΓstar, T, To, R)
```

Comment:
Seeing that you also need to compute these here, I am almost leaning towards having a FarquharModel and an OptimalityFarquharModel, with two methods for `compute_photosynthesis`. 

Perhaps we can discuss with Renato what would make the most sense to him, since he has experience with these models. maybe next week?